pkg/generate/app/sourcelookup.go

```diff
@@ -44,45 +44,32 @@ type SourceRepository struct {
 // NewSourceRepository creates a reference to a local or remote source code repository from
 // a URL or path.
 func NewSourceRepository(s string) (*SourceRepository, error) {
-	var location *url.URL
-	switch {
-	case strings.HasPrefix(s, "git@"):
-		base := "git://" + strings.TrimPrefix(s, "git@")
-		url, err := url.Parse(base)
-		if err != nil {
-			return nil, err
-		}
-		location = url
+	uri, err := url.Parse(s)
```

Comment:
Do you have that laid out?  Because I don't want to endorse an API design until I see that.

----- Original Message -----
> > @@ -44,45 +44,32 @@ type SourceRepository struct {
> >  // NewSourceRepository creates a reference to a local or remote source
> >  code repository from
> >  // a URL or path.
> >  func NewSourceRepository(s string) (*SourceRepository, error) {
> > -	var location *url.URL
> > -	switch {
> > -	case strings.HasPrefix(s, "git@"):
> > -		base := "git://" + strings.TrimPrefix(s, "git@")
> > -		url, err := url.Parse(base)
> > -		if err != nil {
> > -			return nil, err
> > -		}
> > -		location = url
> > +	uri, err := url.Parse(s)
> 
> Yes, I'm aware of that and the token, but my
> [card](https://trello.com/c/KnGYg1Kn/286-5-integrate-with-external-git-solutions-scm-devexp-codefreeze)
> covers just ssh for now. I looked into others and I do have a solution more
> or less in my head already :wink:
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1926/files#r29573706